src/hooks/usePrivatePersonalDetails.ts

```diff
@@ -1,19 +1,20 @@
 import {useContext, useEffect} from 'react';
-import _ from 'underscore';
 import {NetworkContext} from '@components/OnyxProvider';
 import * as PersonalDetails from '@userActions/PersonalDetails';
 
 /**
  * Hook for fetching private personal details
  */
 export default function usePrivatePersonalDetails() {
-    const {isOffline} = useContext(NetworkContext);
+    const network = useContext(NetworkContext);
 
     useEffect(() => {
         const personalDetails = PersonalDetails.getPrivatePersonalDetails();
-        if (isOffline || (Boolean(personalDetails) && !_.isUndefined(personalDetails.isLoading))) {
+        // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+        if (network?.isOffline || (Boolean(personalDetails) && personalDetails?.isLoading !== undefined)) {
             return;
         }
+
         PersonalDetails.openPersonalDetailsPage();
-    }, [isOffline]);
+    }, [network]);
```

Comment:
I would avoid passing objects to dependency array

Reply:
Updated